diff --git a/dist/compiler/plugins/mdx.js b/dist/compiler/plugins/mdx.js
index 651ba9dfe532b84c3f74e47c20e7b71b6b34e0dd..0184a5ad6ef5b1c68aeb2503f5d96537ec6bea2d 100644
--- a/dist/compiler/plugins/mdx.js
+++ b/dist/compiler/plugins/mdx.js
@@ -96,7 +96,8 @@ export const handle = typeof attributes !== "undefined" && attributes.handle;
             pragma: "React.createElement",
             pragmaFrag: "React.Fragment",
             rehypePlugins,
-            remarkPlugins
+            remarkPlugins,
+            providerImportSource: "@mdx-js/react"
           });
           let contents = `
 ${compiled.value}
