diff --git a/dist/compiler/plugins/mdx.js b/dist/compiler/plugins/mdx.js
index ed4befeed2bad93957a45f96c3e2cbd6ead31829..e7db6627384e28e3ae4f9494b322720b81c58730 100644
--- a/dist/compiler/plugins/mdx.js
+++ b/dist/compiler/plugins/mdx.js
@@ -102,7 +102,8 @@ export const meta = typeof attributes !== "undefined" && attributes.meta;
             pragma: "React.createElement",
             pragmaFrag: "React.Fragment",
             rehypePlugins,
-            remarkPlugins
+            remarkPlugins,
+            providerImportSource: "@mdx-js/react",
           });
           let contents = `
 ${compiled.value}
