diff --git a/dist/compiler/plugins/mdx.js b/dist/compiler/plugins/mdx.js
index 727fa32c7ff0db6fd1db6ab81aa9c6d9d819d4a4..51ab32cb5a62dfe54883d7004f5c3499f3dc2422 100644
--- a/dist/compiler/plugins/mdx.js
+++ b/dist/compiler/plugins/mdx.js
@@ -96,7 +96,8 @@ export const handle = typeof attributes !== "undefined" && attributes.handle;
             pragma: "React.createElement",
             pragmaFrag: "React.Fragment",
             rehypePlugins,
-            remarkPlugins
+            remarkPlugins,
+            providerImportSource: "@mdx-js/react"
           });
           let contents = `
 ${compiled.value}
