diff --git a/node_modules/@lerna/project/index.js b/node_modules/@lerna/project/index.js
index 789f37b..4a48f87 100644
--- a/node_modules/@lerna/project/index.js
+++ b/node_modules/@lerna/project/index.js
@@ -166,11 +166,13 @@ class Project {
     return finder;
   }
 
-  getPackages() {
-    const mapper = packageConfigPath =>
-      loadJsonFile(packageConfigPath).then(
-        packageJson => new Package(packageJson, path.dirname(packageConfigPath), this.rootPath)
+  getPackages(redirect = path => path) {
+    const mapper = packageJsonPath => {
+      packageJsonPath = redirect(packageJsonPath);
+      return loadJsonFile(packageJsonPath).then(
+        packageJson => new Package(packageJson, path.dirname(packageJsonPath), this.rootPath)
       );
+    };
 
     return this.fileFinder("package.json", filePaths => pMap(filePaths, mapper, { concurrency: 50 }));
   }
